packages/react-query/src/__tests__/useSequentialMutations.test.tsx

```
@@ -0,0 +1,1236 @@
+    }
+
+    const r = renderWithClient(queryClient, <Page />)
+    expect(r.getByText('pending')).toBeInTheDocument()
+    await vi.advanceTimersByTimeAsync(6)
+    expect(r.getByText('X-done')).toBeInTheDocument()
+  })
+
+  it('uses getServerSnapshot during SSR render (len reflects observers)', () => {
+    // render on server to trigger getServerSnapshot path of useSyncExternalStore
+    const { renderToString } =
+      require('react-dom/server') as typeof import('react-dom/server')
```

ESLint

[error] `import()` type annotations are forbidden.

(@typescript-eslint/consistent-type-imports)